physx/src/foundation.rs

```diff
@@ -281,3 +313,42 @@ unsafe impl AllocatorCallback for DefaultAllocator {
         unreachable!()
     }
 }
+
+unsafe extern "C" fn report_error_helper(
+    code: u32,
+    message: *const c_void,
+    file: *const c_void,
+    line: u32,
+    user_data: *const c_void,
+) {
+    let code = BitFlags::from_bits(code).unwrap_or_else(|_| {
+        debug_assert!(false, "bad error code {}", code);
+        Default::default()
+    });
+    let message =
+        String::from_utf8_lossy(unsafe { CStr::from_ptr(message.cast::<i8>()) }.to_bytes());
+    let file = String::from_utf8_lossy(unsafe { CStr::from_ptr(file.cast::<i8>()) }.to_bytes());
+    let ec = unsafe { &*user_data.cast::<Box<dyn ErrorCallback>>() };
+    ec.report_error(code, &message, &file, line);
+}
+
+unsafe extern "C" fn error_userdata_drop_helper(user_data: *mut c_void) {
+    let b: Box<Box<dyn ErrorCallback>> =
```

Comment:
some funky stuff here. but it should be correct right?

Reply:
See my comments above - not sure if we care about dealloc, but this doesn't feel good.